java-connectors/kafka-connect-azure-servicebus/src/main/java/io/lenses/streamreactor/connect/azure/servicebus/source/TaskToReceiverBridge.java

```diff
@@ -0,0 +1,156 @@
+   */
+  public TaskToReceiverBridge(Map<String, String> properties,
+      BlockingQueue<ServiceBusMessageHolder> recordsQueue, OffsetStorageReader offsetStorageReader) {
+    this.recordsQueue = recordsQueue;
+    this.offsetProvider = new ServiceBusPartitionOffsetProvider(offsetStorageReader);
+    String connectionString = properties.get(AzureServiceBusConfigConstants.CONNECTION_STRING);
+    List<Kcql> kcqls =
+        KcqlConfigBusMapper.mapKcqlsFromConfig(properties.get(AzureServiceBusConfigConstants.KCQL_CONFIG));
+    sentMessagesExecutors = Executors.newFixedThreadPool(kcqls.size() * 2);
+    recordsToCommitMap = new ConcurrentHashMap<>(INITIAL_RECORDS_TO_COMMIT_SIZE);
+
+    initiateReceivers(recordsQueue, kcqls, connectionString);
+  }
+
+  /**
+   * Creates Bridge between Receivers and Connector's Task for Azure Service Bus.
+   *
+   * @param connectionString    Service Bus connection string
+   * @param kcqls               list of KCQLs to initiate Receivers for.
+   * @param recordsQueue        records queue used to store received messages.
+   * @param offsetStorageReader offset storage reader from Task.
+   */
+  public TaskToReceiverBridge(String connectionString, List<Kcql> kcqls,
+      BlockingQueue<ServiceBusMessageHolder> recordsQueue, OffsetStorageReader offsetStorageReader) {
+    this.recordsQueue = recordsQueue;
+    this.offsetProvider = new ServiceBusPartitionOffsetProvider(offsetStorageReader);
+    sentMessagesExecutors = Executors.newFixedThreadPool(kcqls.size() * 2);
+    recordsToCommitMap = new ConcurrentHashMap<>(INITIAL_RECORDS_TO_COMMIT_SIZE);
+
+    initiateReceivers(recordsQueue, kcqls, connectionString);
+  }
+
+  /**
+   * Creates Bridge between Receivers and Connector's Task for Azure Service Bus.
+   *
+   * @param recordsQueue          records queue used to store received messages.
+   * @param offsetStorageReader   offset storage reader from Task.
+   * @param receivers             map of {@link ServiceBusReceiverFacade} receivers.
+   * @param sentMessagesExecutors {@link ExecutorService} that handles sent messages.
+   */
+  TaskToReceiverBridge(BlockingQueue<ServiceBusMessageHolder> recordsQueue,
+      OffsetStorageReader offsetStorageReader, Map<String, ServiceBusReceiverFacade> receivers,
+      ExecutorService sentMessagesExecutors) {
+    this.recordsQueue = recordsQueue;
+    this.offsetProvider = new ServiceBusPartitionOffsetProvider(offsetStorageReader);
+    this.sentMessagesExecutors = sentMessagesExecutors;
+    this.receivers = receivers;
+    recordsToCommitMap = new ConcurrentHashMap<>(INITIAL_RECORDS_TO_COMMIT_SIZE);
+  }
+
+  private void initiateReceivers(BlockingQueue<ServiceBusMessageHolder> recordsQueue,
+      List<Kcql> kcqls, String connectionString) {
+    receivers = new ConcurrentHashMap<>(kcqls.size());
+
+    kcqls.stream().forEach(kcql -> {
+      String receiverId = ServiceBusReceiverFacade.class.getSimpleName() + UUID.randomUUID();
```

Comment:
`ServiceBusReceiverFacade.class.getSimpleName()`
this should be extracted to a static final constant at the class level as it is (hopefully!) not going to change during runtime(!)

Reply:
Good spot, removed!